.github/actions/build.sh

```diff
@@ -58,8 +58,8 @@ do
 
     if [ "${GH_ARCH%%-*}" != "noarch" ]; then
         # use TCVERSION and ARCH to get real exit codes.
-        echo "$ make TCVERSION=${GH_ARCH##*-} ARCH=${GH_ARCH%%-*} -C ./spk/${package}" >>build.log
-        make TCVERSION=${GH_ARCH##*-} ARCH=${GH_ARCH%%-*} -C ./spk/${package} ${MAKE_ARGS} |& tee >(tail -15 >>build.log)
+        echo "$ make arch-${GH_ARCH%%-*}-${GH_ARCH##*-} -C ./spk/${package}" >>build.log
+        make arch-${GH_ARCH%%-*}-${GH_ARCH##*-} -C ./spk/${package} ${MAKE_ARGS} |& tee >(tail -15 >>build.log)
```

Comment:
Sadly this will not work. My experience was that only `make TCVERSION=...  ARCH=...` returns the correct exit status as my comment above indicates.

Reply:
Then I'll have to see what can be done to alleviate this... My guess is that `-@MAKEFLAGS= ` invokes a sub-shell and thus you just can't track process status.  There may be something to do about it.

Here's the situation (this really has been a very long-term goal and I think I'm now really close):
- Objective: Provide kernel module packages (cdrom, usb-dvb, usb-wifi, etc).
- Howto: Provide DSM6 and DSM7 packages that contains modules for all corresponding kernel versions such as, a DSM-6.1 `aarch64` package contains:
```
lib/modules/armada37xx-6.2.2/
lib/modules/armada37xx-6.2.2/4.4.59+/
lib/modules/armada37xx-6.2.2/4.4.59+/drivers/
lib/modules/armada37xx-6.2.2/4.4.59+/drivers/cdrom/
lib/modules/armada37xx-6.2.2/4.4.59+/drivers/cdrom/cdrom.ko
lib/modules/armada37xx-6.2.2/4.4.59+/drivers/scsi/
lib/modules/armada37xx-6.2.2/4.4.59+/drivers/scsi/sr_mod.ko
lib/modules/armada37xx-6.2.3/
lib/modules/armada37xx-6.2.3/4.4.59+/
lib/modules/armada37xx-6.2.3/4.4.59+/drivers/
lib/modules/armada37xx-6.2.3/4.4.59+/drivers/cdrom/
lib/modules/armada37xx-6.2.3/4.4.59+/drivers/cdrom/cdrom.ko
lib/modules/armada37xx-6.2.3/4.4.59+/drivers/scsi/
lib/modules/armada37xx-6.2.3/4.4.59+/drivers/scsi/sr_mod.ko
lib/modules/rtd1296-6.1/
lib/modules/rtd1296-6.1/4.4.15+/
lib/modules/rtd1296-6.1/4.4.15+/drivers/
lib/modules/rtd1296-6.1/4.4.15+/drivers/cdrom/
lib/modules/rtd1296-6.1/4.4.15+/drivers/cdrom/cdrom.ko
lib/modules/rtd1296-6.1/4.4.15+/drivers/scsi/
lib/modules/rtd1296-6.1/4.4.15+/drivers/scsi/sr_mod.ko
lib/modules/rtd1296-6.2/
lib/modules/rtd1296-6.2/4.4.59+/
lib/modules/rtd1296-6.2/4.4.59+/drivers/
lib/modules/rtd1296-6.2/4.4.59+/drivers/cdrom/
lib/modules/rtd1296-6.2/4.4.59+/drivers/cdrom/cdrom.ko
lib/modules/rtd1296-6.2/4.4.59+/drivers/scsi/
lib/modules/rtd1296-6.2/4.4.59+/drivers/scsi/sr_mod.ko
lib/modules/rtd1296-6.2.2/
lib/modules/rtd1296-6.2.2/4.4.59+/
lib/modules/rtd1296-6.2.2/4.4.59+/drivers/
lib/modules/rtd1296-6.2.2/4.4.59+/drivers/cdrom/
lib/modules/rtd1296-6.2.2/4.4.59+/drivers/cdrom/cdrom.ko
lib/modules/rtd1296-6.2.2/4.4.59+/drivers/scsi/
lib/modules/rtd1296-6.2.2/4.4.59+/drivers/scsi/sr_mod.ko
lib/modules/rtd1296-6.2.3/
lib/modules/rtd1296-6.2.3/4.4.59+/
lib/modules/rtd1296-6.2.3/4.4.59+/drivers/
lib/modules/rtd1296-6.2.3/4.4.59+/drivers/cdrom/
lib/modules/rtd1296-6.2.3/4.4.59+/drivers/cdrom/cdrom.ko
lib/modules/rtd1296-6.2.3/4.4.59+/drivers/scsi/
lib/modules/rtd1296-6.2.3/4.4.59+/drivers/scsi/sr_mod.ko
```

This mean that package is kernel-binary compatible to what-ever DSM you are running (except 6.2.4 which hasn't been released just yet, but in many cases previous version will still work).  All is left is for you to insmod the kernel-corresponding modules as you require.

But for this to work I adapted `spksrc.spk.mk` in order to loop over the sub-archs for generic packages.  There is no way I can do this otherwise which mean that current github-action calls must be converted to spksrc default make behavior for binary packages, at the very least for generic archs.

At first glance I see two options:
1. fix the underlying issue so all calls can be converted to `make arch-<arch>-<version>`
2. add a check for kernel module compiling and only then use `make arch-<arch>-<version>` calls.

Reply:
@hgy59 all issues seems to be fixed now with the extra workaround I added (e.g. checking file result existence).  Besides adding an icon to the package I believe this is ready for your review (as there are a few framework changes).  Thnx in advance.